orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/support/TargetServerGroupLinearStageSupport.groovy

```diff
@@ -34,50 +36,55 @@ abstract class TargetServerGroupLinearStageSupport extends LinearStage implement
 
   String name = this.type
 
-  TargetServerGroupLinearStageSupport(String name) {
-    super(name)
+  @Override
+  def <T extends Execution> List<Stage<T>> aroundStages(Stage<T> parentStage) {
```

Comment:
This logic necessitated a move away from separate `preStages`/`postStages` methods ... although it's probably possible to refactor the creation logic such that we could do a definitive split into pre/post.

Given that these stages include a `SyntheticStageOwner` indicating whether they're `STAGE_BEFORE` or `STAGE_AFTER`, I'm not sure whether a further split into methods is necessary.

That said, I'm not convinced we should be returning concrete `Stage` implementations here.